plugins/bulk-import/src/hooks/useRepositories.ts

```diff
@@ -46,83 +71,127 @@ export const useRepositories = (options: {
+                  ),
+                },
+              },
+            };
+          }, {}) || {};
+        setRepositoriesData(repoData);
+        setTotalRepositories(result?.totalCount);
+        setIsLoading(false);
+      };
+      const prepareDataForOrganizations = () => {
+        const orgData: { [id: string]: AddRepositoryData } =
+          result?.organizations?.reduce(
+            (acc: { [id: string]: AddRepositoryData }, val: Repository) => {
+              return {
+                ...acc,
+                [val.id]: {
+                  id: val.id,
+                  orgName: val.name,
+                  organizationUrl: `https://github.com/${val?.name}`,
+                  totalReposInOrg: val.totalRepoCount,
+                },
+              };
+            },
+            {},
+          ) || {};
+        setOrganizationsData(orgData);
+        setTotalOrganizations(result?.totalCount);
+        setIsLoading(false);
+      };
+      if (!dataLoading)
+        if (options?.showOrganizations) {
+          prepareDataForOrganizations();
+        } else {
+          prepareDataForRepositories();
+        }
+    },
+    [
+      user,
+      options?.showOrganizations,
+      baseUrl,
+      setRepositoriesData,
+      setOrganizationsData,
+      setTotalOrganizations,
+      setTotalRepositories,
+    ],
+  );
+
+  const debouncedUpdateResources = useDebounceCallback(prepareData, 200);
+
   React.useEffect(() => {
-    const prepareDataForRepositories = () => {
-      const repoData: AddRepositoryData[] =
-        value?.repositories?.map((val: Repository) => ({
-          id: val.id || `${val.organization}/${val.name}`,
-          repoName: val.name,
-          defaultBranch: val.defaultBranch,
-          orgName: val.organization,
-          repoUrl: val.url,
-          organizationUrl: val?.url?.substring(
-            0,
-            val.url.indexOf(val?.name || '') - 1,
-          ),
-          catalogInfoYaml: {
-            prTemplate: getPRTemplate(
-              val.name || '',
-              val.organization || '',
-              user as string,
-              baseUrl as string,
-              val.url || '',
-              val.defaultBranch || 'main',
-            ),
-          },
-        })) || [];
-      setRepositoriesData(repoData);
-    };
-    const prepareDataForOrganizations = () => {
-      const orgData: AddRepositoryData[] =
-        value?.organizations?.map((val: Repository) => ({
-          id: val.id,
-          orgName: val.name,
-          organizationUrl: `https://github.com/${val?.name}`,
-          totalReposInOrg: val.totalRepoCount,
-        })) || [];
-      setOrganizationsData(orgData);
-    };
-    if (options?.showOrganizations) {
-      prepareDataForOrganizations();
-    } else {
-      prepareDataForRepositories();
-    }
-  }, [value, user, options?.showOrganizations, baseUrl]);
+    debouncedUpdateResources?.(value, loading);
+  }, [debouncedUpdateResources, value, loading, error]);
```

Comment:
`error` in dependency array but not used in effect, see if this can be avoided or is it required?